testnet/stacks-node/src/neon_node.rs

```diff
@@ -2255,6 +2262,151 @@ impl BlockMinerThread {
+            StackerDBSession::new(&self.config.node.rpc_bind, miner_contract_id);
+        let miner_slot_ids: Vec<_> = (0..MINER_SLOT_COUNT * 2).collect();
+        if let Ok(messages) = miners_stackerdb.get_latest_chunks(&miner_slot_ids) {
+            for message in messages {
+                if let Some(message) = message {
+                    if message.is_empty() {
+                        continue;
+                    }
+                    let Ok(SignerMessage::MockMinerMessage(miner_message)) =
+                        SignerMessage::consensus_deserialize(&mut message.as_slice())
+                    else {
+                        continue;
+                    };
+                    if miner_message.tenure_burn_block_height == self.burn_block.block_height {
+                        debug!(
+                            "Already sent mock miner message for tenure burn block height {:?}",
+                            self.burn_block.block_height
+                        );
+                        return Ok(());
+                    }
+                }
+            }
+        }
+        // Retrieve any MockSignatures from stackerdb
+        let mut mock_signatures = Vec::new();
+        let reward_cycle = self
+            .burnchain
+            .block_height_to_reward_cycle(self.burn_block.block_height)
+            .expect("BUG: block commit exists before first block height");
+        let signers_contract_id = MessageSlotID::MockSignature
+            .stacker_db_contract(self.config.is_mainnet(), reward_cycle);
+        // Get the slots for every signer
+        let stackerdbs = StackerDBs::connect(&self.config.get_stacker_db_file_path(), false)?;
+        let slot_ids: Vec<_> = stackerdbs
+            .get_signers(&signers_contract_id)
+            .expect("FATAL: could not get signers from stacker DB")
+            .into_iter()
+            .enumerate()
+            .map(|(slot_id, _)| {
+                u32::try_from(slot_id).expect("FATAL: too many signers to fit into u32 range")
+            })
+            .collect();
+        let chunks = stackerdbs.get_latest_chunks(&signers_contract_id, &slot_ids)?;
+        for chunk in chunks {
+            if let Some(chunk) = chunk {
+                if let Ok(SignerMessage::MockSignature(mock_signature)) =
+                    SignerMessage::consensus_deserialize(&mut chunk.as_slice())
+                {
+                    if mock_signature.sign_data.event_burn_block_height
+                        == self.burn_block.block_height
+                    {
+                        mock_signatures.push(mock_signature);
+                    }
+                }
+            }
+        }
+
+        let server_version = version_string(
+            "stacks-node",
+            option_env!("STACKS_NODE_VERSION")
+                .or(option_env!("CARGO_PKG_VERSION"))
+                .unwrap_or("0.0.0.0"),
+        );
+        let stacks_tip_height = p2p_net.stacks_tip.height;
```

Comment:
This information -- `stacks_tip`, `burnchain_tip`, and `chain_view` -- can be synthesized through queries against the Stacks chainstate DB and sortition DB.